src/drt/src/ta/FlexTA_init.cpp

```diff
@@ -235,21 +247,32 @@ void FlexTAWorker::initIroute_helper_generic_helper(frGuide* guide,
   frMTerm* trueTerm = nullptr;
   for (auto& term : result) {
     frInst* inst = nullptr;
-    if (term->typeId() == frcInstTerm) {
-      if (static_cast<frInstTerm*>(term)->getNet() != net) {
-        continue;
+    switch (term->typeId()) {
+      case frcInstTerm: {
+        auto iterm = static_cast<frInstTerm*>(term);
+        if (iterm->getNet() != net) {
+          continue;
+        }
+        inst = iterm->getInst();
+        inst->getTransform(shiftXform);
+        shiftXform.setOrient(dbOrientType(dbOrientType::R0));
+        inst->getUpdatedXform(instXform);
+        trueTerm = iterm->getTerm();
+        break;
       }
-      inst = static_cast<frInstTerm*>(term)->getInst();
-      inst->getTransform(shiftXform);
-      shiftXform.setOrient(dbOrientType(dbOrientType::R0));
-      inst->getUpdatedXform(instXform);
-      trueTerm = static_cast<frInstTerm*>(term)->getTerm();
-    } else if (term->typeId() == frcTerm) {
-      if (static_cast<frTerm*>(term)->getNet() != net) {
-        continue;
+      case frcBTerm:
+      case frcMTerm: {
+        /*auto mterm = static_cast<frMTerm*>(term);
+        if (mterm->getNet() != net) {
+          continue;
+        }
+        trueTerm = mterm;
+        break;*/
```

Comment:
is this commented out intentionally?

Reply:
Yes, per the email thread we had, this code doesn't ultimately do anything. It looks like a bug of some sort so I'm leaving it in for reference if someone can figure it out.